chain-tx-enclave-next/mls/src/astree.rs

```diff
@@ -173,16 +172,14 @@ impl<D: Input + BlockInput + FixedOutput + Reset + Default + Clone> GroupKeySour
         sender_leaf: LeafSize,
         cs: CipherSuite,
     ) -> (SecretVec<u8>, Vec<u8>) {
-        if self.ratchets[sender_leaf.0].is_none() {
+        if self.ratchets[sender_leaf.node_index()].is_none() {
             let base_secret = self
                 .base_secret_source
                 .get_base_secret(group_context_hash.clone(), sender_leaf);
-            self.ratchets[sender_leaf.0] = Some(HashRatchet::new(
-                NodeSize::from_leaf_index(sender_leaf),
-                base_secret,
-            ));
+            self.ratchets[sender_leaf.0 as usize] =
+                Some(HashRatchet::new(NodeSize::from(sender_leaf), base_secret));
         };
-        let ratchet = self.ratchets[sender_leaf.0].as_mut().unwrap();
+        let ratchet = self.ratchets[sender_leaf.0 as usize].as_mut().unwrap();
```

Comment:
expect()?

Reply:
it's set to `Some` immediately previous line of code.

Reply:
should those two be `self.ratchets[sender_leaf.node_index()]` instead of `self.ratchets[sender_leaf.0 as usize]`?